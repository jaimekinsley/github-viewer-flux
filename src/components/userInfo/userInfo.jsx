import React from 'react';
import { useFetchUser, useState } from '../../hooks/appContext';

const UserInfo = () => {
  const  user = useFetchUser();

  if(!user) return <h3>Enter username</h3>;

  return (
    <>
      <p>Name: {user.name}</p>
      <p>Number of Followers: {user.followers}</p>
      <p>Number of Following: {user.following}</p>
      <p>GitHub Profile: <a href= {user.html_url}>{user.html_url}</a></p>
    </>
  );
};

export default UserInfo;
