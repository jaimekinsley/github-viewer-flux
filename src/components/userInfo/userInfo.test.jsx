import React from 'react';
import { shallow } from 'enzyme';
import userInfo from './userInfo';

describe('Form component', () => {
  it('renders UserInfo', () => {
    const wrapper = shallow(<userInfo user={{
      name: 'Jaime Sanders',
      followers: 5,
      following: 3,
      html_url: 'http://github.com/jaimekinsley'
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
});

