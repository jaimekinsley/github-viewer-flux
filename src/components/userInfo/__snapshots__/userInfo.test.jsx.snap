// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders UserInfo 1`] = `
<userInfo
  user={
    Object {
      "followers": 5,
      "following": 3,
      "html_url": "http://github.com/jaimekinsley",
      "name": "Jaime Sanders",
    }
  }
/>
`;
